src/translations/LanguageContext.tsx

```diff
@@ -19,35 +20,62 @@ const AppLanguageProvider: React.FC = ({children}) => {
 function useLanguage() {
   const [currentLanguage, setCurrentLanguage] = useState(DEFAULT_LANGUAGE);
   const [locale, setLocale] = useState(preferredLocale);
+  const {
+    preferences: {useSystemLanguage, language: userPreferencedLanguage},
+  } = usePreferences();
+
+  // Usage of system setting not explicitly chosen, and user preferenced language has value
+  const shouldUseUserPreferenced =
+    !useSystemLanguage && !!userPreferencedLanguage;
+
+  const getAndSetPreferencedLanguage = () => {
+    if (shouldUseUserPreferenced) {
+      const newLanguage = getAsAppLanguage(userPreferencedLanguage);
+      setCurrentLanguage(newLanguage);
+    }
+  };
 
   useEffect(() => {
-    const onChange = () => {
-      setLocale(preferredLocale);
-    };
-    RNLocalize.addEventListener('change', onChange);
-    return () => {
-      RNLocalize.removeEventListener('change', onChange);
-    };
+    if (shouldUseUserPreferenced) {
+      getAndSetPreferencedLanguage();
+    } else {
+      const onChange = () => {
+        setLocale(preferredLocale);
+      };
+      RNLocalize.addEventListener('change', onChange);
+      return () => {
+        RNLocalize.removeEventListener('change', onChange);
+      };
+    }
   }, []);
```

Comment:
Shouldn't `shouldUseUserPreferenced` be in the dependency list here? Probably a corner case, but from my understanding the localization change listener is now only added if `shouldUseUserPreferenced` is false at initial mounting, and not if the user changes their setting to using the system language.

Adding `shouldUseUserPreferenced` will also make the `useEffect` on line 61 redundant.

Reply:
Of course, not sure what I was even thinking. Good catch!

Reply:
I think maybe it's time to include dependency list eslint rule. I've also had some issues remembering all deps 😅